src/clr-angular/data/datagrid/_datagrid.clarity.scss

```diff
@@ -3,266 +3,271 @@
 
 @include exports('datagrid.clarity') {
-    @include basic-table(".datagrid", ".datagrid-head", ".datagrid-body",
-        ".datagrid-row", ".datagrid-column", ".datagrid-cell");
-
-    $clr-datagrid-icon-size: 0.583333rem;
-    $clr-datagrid-action-arrow-size: 0.25rem;
-    $clr-datagrid-fixed-column-size: (2 * $clr-table-cellpadding) + $clr-table-lineheight;
-    $clr-datagrid-column-handle-tracker-top: -0.5rem;
-    $clr-datagrid-column-separator-expandby: 0.5rem;
-    $clr-datagrid-compact-outline-offset: -0.1666667rem;
+    @include basic-table(".datagrid", ".datagrid-header", ".datagrid-body",
+            ".datagrid-row", ".datagrid-column", ".datagrid-cell");
 
     .datagrid-host {
         display: flex;
         flex-flow: column nowrap;
-    }
-
-    .datagrid-overlay-wrapper {
-        display: flex;
-        flex-direction: row;
-        flex: 0 auto; // Same as flex: initial. It sizes the item based on width/height properties or its content.
+        height: auto;
+        position: relative; // for the .datagrid-spinner to work.
         width: 100%;
-        min-height: 100%; // Must be min to fix Safari bug: height: 100% won't fill flex parent. http://bit.ly/2qAPkha
-        overflow-x: auto;
-        overflow-y: hidden;
-
-        .datagrid-spinner {
-            flex: 0 0 100%;
-            margin-left: -100%;
-            display: flex;
-            justify-content: center;
-            align-items: center;
-            background: $clr-datagrid-loading-background;
-        }
-
-    }
-    .datagrid-scroll-wrapper {
-        display: flex;
-        flex-direction: row;
-        min-width: 100%;
     }
+
     .datagrid {
         display: flex;
-        flex-flow: column nowrap;
+        flex-wrap: wrap;
+        flex: 1 1 auto;
+        width: 100%;
+        margin-top: 0;
+        overflow-x: auto; // Enables x scrolling when columns are wider than the datagrid.
+        // Changed to scroll b/c I saw double scroll bars w/ auto in firefox.
         // IE & Firefox needs this
-        min-height: 1px;
+        min-height: 1px; // Do we still need this? It was ported over
+        // Overrides the table defaults so the new structure has correct corners
+        border-bottom-left-radius: 0;
+        border-bottom-right-radius: 0;
+
+        -ms-overflow-style: -ms-autohiding-scrollbar;
     }
+
     .datagrid-head, .datagrid-body, .datagrid-row, .datagrid-column, .datagrid-cell, .datagrid-fixed-column {
         display: block;
     }
-    .datagrid-table-wrapper {
-        flex: 1 1 auto;
-        display: flex;
-        flex-flow: column nowrap;
-        // IE & Firefox needs this
-        min-height: 1px;
-    }
-    .datagrid-body {
-        flex: 1 1 auto;
-        overflow-y: auto;
-        -ms-overflow-style: -ms-autohiding-scrollbar;
 
-        display: flex;
-        flex-direction: column;
+    .datagrid-body .datagrid-row {
```

Comment:
Do we need to nest the `.datagrid-row` selector under `.datagrid-body`? Is there a case where the `.datagrid-row` is present outside of the `.datagrid-body`? If not, just `.datagrid-row` should be fine here.

Reply:
This was copied over from the original stylesheet. I moved it out and it seems like the row components are ok. Will move it out from under .datagrid-body.

Will update.